Client/Frontend/Intro/IntroViewControllerV2.swift

```diff
@@ -0,0 +1,108 @@
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import UIKit
+import SnapKit
+import Shared
+import Leanplum
+
+class IntroViewControllerV2: UIViewController {
+    private lazy var welcomeCard: IntroScreenWelcomeView = {
+        let welcomeCardView = IntroScreenWelcomeView()
+        welcomeCardView.contentMode = .scaleAspectFit
+        welcomeCardView.clipsToBounds = true
+        welcomeCardView.tag = 0
+        return welcomeCardView
+    }()
+    private lazy var syncCard: IntroScreenSyncView = {
+        let syncCardView = IntroScreenSyncView()
+        syncCardView.contentMode = .scaleAspectFit
+        syncCardView.clipsToBounds = true
+        syncCardView.tag = 0
+        return syncCardView
+    }()
+    // Closure delegate
+    var didFinishClosure: ((IntroViewControllerV2,FxAPageType?) -> Void)?
+    
+    // MARK: Initializer
+    init() {
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        initialViewSetup()
+    }
+    
+    // MARK: View setup
+    private func initialViewSetup() {
+        // Initialize
+        view.addSubview(syncCard)
+        view.addSubview(welcomeCard)
+        // Constraints
+        setupWelcomeCard()
+        setupSyncCard()
+    }
+
+    private func setupWelcomeCard() {
+        // Constraints
+        welcomeCard.snp.makeConstraints { make in
+            make.edges.equalToSuperview()
+        }
+        // Buton action closures
+        // Next button action
+        welcomeCard.nextClosure = {
+            print("Going to next screen")
+            UIView.animate(withDuration: 0.3, animations: {
+                self.welcomeCard.alpha = 0
+            }) { _ in
+                self.welcomeCard.isHidden = true
+            }
+        }
+        // Close button action
+        welcomeCard.closeClosure = {
+            self.didFinishClosure?(self, nil)
+        }
+    }
+    
+    private func setupSyncCard() {
+        syncCard.snp.makeConstraints() { make in
+            make.edges.equalToSuperview()
+        }
+        // Start browsing button action
+        syncCard.startBrowsing = {
+            self.didFinishClosure?(self, nil)
+        }
+        // Sign-up browsing button action
+        syncCard.signUp = {
+            self.didFinishClosure?(self, .signUpFlow)
+        }
+    }
+}
+
+// MARK: UIViewController setup
+extension IntroViewControllerV2 {
+    override var prefersStatusBarHidden: Bool {
+        return true
+    }
+
+    override var shouldAutorotate: Bool {
+        return false
+    }
+
+    override var supportedInterfaceOrientations: UIInterfaceOrientationMask {
```

Comment:
**Refactoring Note:** We use portrait mode in certain VCs and was thinking of making a higher level PortrainModeViewController class to have the default override methods.

Reply:
Yeah, maybe we can commonize there. Wondering if we can avoid inheritance, and encourage mix-in object-oriented design (because I keep coming across similar cases like this where I really want mix-in behaviour and not the 'weight' of inheritance) and do something like:

```
// You need to return these for fixing VC to portrait mode
struct PortraitModeVCConstants {
  static let prefersStatusBarHidden = false
  static let shouldAutorotate = false
  static let supportedInterfaceOrientations = false
}

extension IntroViewControllerV2 {
    override var prefersStatusBarHidden: Bool {
        return PortraitModeVCConstants.prefersStatusBarHidden
    }
  (ETC...)
```
As we aren't trying to save typing, just want to centralize the logic in these cases.